File: src/main/java/app/askresume/api/access/validator/OauthValidator.java
package app.askresume.api.access.validator;

import app.askresume.domain.member.constant.MemberType;
import app.askresume.global.error.ErrorCode;
import app.askresume.global.error.exception.BusinessException;
import org.springframework.stereotype.Service;

@Service
public class OauthValidator {

    public void validateMemberType(String memberType) {
        if(!MemberType.isMemberType(memberType)) {
            throw new BusinessException(ErrorCode.INVALID_MEMBER_TYPE);
        }
    }

}

File: src/main/kotlin/app/askresume/api/member/validator/OauthValidator.kt
package app.askresume.api.member.validator

import app.askresume.domain.member.constant.MemberType
import app.askresume.global.error.ErrorCode
import app.askresume.global.error.exception.BusinessException
import org.springframework.stereotype.Service

@Service
class OauthValidator {

    fun validateMemberType(memberType: String) {
        if (!MemberType.isMemberType(memberType)) {
            throw BusinessException(ErrorCode.INVALID_MEMBER_TYPE)
        }
    }

}

